core/rust/qdbr/src/parquet_read/meta.rs

```diff
@@ -1,30 +1,39 @@
 use crate::parquet_read::{ColumnChunkBuffers, ColumnMeta, ParquetDecoder};
+use crate::parquet_write::schema::ColumnType;
+use crate::parquet_write::QDB_TYPE_META_PREFIX;
 use parquet2::metadata::Descriptor;
 use parquet2::read::read_metadata;
 use parquet2::schema::types::PrimitiveLogicalType::{Timestamp, Uuid};
 use parquet2::schema::types::{
     IntegerType, PhysicalType, PrimitiveConvertedType, PrimitiveLogicalType, TimeUnit,
 };
+use std::collections::HashMap;
 use std::fs::File;
 
-use crate::parquet_write::schema::ColumnType;
-
 impl ParquetDecoder {
     pub fn read(mut reader: File) -> anyhow::Result<Self> {
         let metadata = read_metadata(&mut reader)?;
-
         let col_len = metadata.schema_descr.columns().len();
+        let additional_meta: Option<HashMap<String, Option<String>>> =
+            metadata.key_value_metadata.as_ref().map(|vec| {
+                vec.iter()
+                    .map(|kv| (kv.key.to_string(), kv.value.to_owned()))
```

Comment:
do we allow keys duplicates? if not, it's good to change API from Vec to HashMap

Reply:
Should be no dups. I'd expect we do use HashMap since:
```
dditional_meta: Option<HashMap<String, Option<String>>>
```

Reply:
No duplicates if it's us writing the file.